crates/core/src/subscription/module_subscription_actor.rs

```diff
@@ -98,9 +99,11 @@ impl ModuleSubscriptionManager {
     }
 }
 
+type SubscriptionRw = Arc<RwLock<Subscription>>;
```

Comment:
I actually don't think a lock is necessary anymore, the way you have things written.

It's technically not necessary for incremental evaluation, because `broadcast_event_blocking` will block each reducer until `broadcast_commit_event` completes.

It's also technically not necessary for the initial subscription, because `add_subscriber` acquires an exclusive lock on the datastore.

Of course if any of those assumptions become invalid, then we would need it. However, for now I would remove it and just make sure to comment things sufficiently.

Reply:
primary motivation here is not to prevent race condition but to have a `Sync` so that I can share the reference of `Subscription` between threads.
I am new to rust concurrency, would love to know better ways to do it.